src/nvim/eval.c

```diff
@@ -6007,6 +6012,26 @@ static void rettv_dict_alloc(typval_T *rettv)
   ++d->dv_refcount;
 }
 
+/// Clear the all the keys of a Dictionary.
+/// "d" remains a valid empty Dictionary.
+///
+/// @param d The Dictionary to clear
+void dict_clear(dict_T *d) {
```

Comment:
Mutating functions cannot assume anything else because NULL dictionary cannot be put into the target state. But clearing keys out of NULL dictionary is completely possible: you should simply do nothing because NULL dictionary is already empty.